assets/js/modules/ads/components/setup/SetupMainPAX.js

```diff
@@ -79,16 +97,31 @@ export default function SetupMainPAX( { finishSetup } ) {
 	const { setPaxConversionID, setExtCustomerID, submitChanges } =
 		useDispatch( MODULES_ADS );
 
-	const onCompleteSetup = useCallback( async () => {
-		if ( ! paxApp ) {
+	useMount( () => {
+		if ( PAX_SETUP_STEP.FINISHED === showPaxAppStep ) {
+			// If the PAX query param indicates the setup is finished on page load,
+			// set the step back to the PAX launch, as values are only temporarly
+			// saved in state after PAX campaign setup signal is received.
+			setShowPaxAppQueryParam( PAX_SETUP_STEP.LAUNCH );
+		}
+	} );
+
+	const onLaunch = ( app ) => {
+		if ( ! paxAppRef.current ) {
+			paxAppRef.current = app;
+		}
```

Comment:
Do we need to check for an existing value here? Presumably this will only run once, but it's also safe to reassign this I'd think 🤔 

```suggestion
		paxAppRef.current = app;
```

Reply:
Due to some weirdness in PAX app, like `getServices()` not being callable, then working again, had to put few safeguards to ensure some unexpected changes won't happen

Reply:
I don't understand though, `onLaunch` is only ever called once, right?

If `paxAppRef.current` is defined, it means that this code was already called. What bug is the check for `! paxAppRef.current` fixing? This code should not be called if `paxAppRef.current` is set, because it is only set here and this code only runs once.

Can you include a screencast of the bug you experience when this change isn't made? 🤔

If it does need to stay we should leave a comment explaining it, because it's not intuitive that this code runs multiple times.

Reply:
Technically shouldn't make a problem to remove it, tested it now without it, not causing any unexpected errors. It was set as a local state before, when Simon gave it a quick test, app instance was null in campaign created callback, although `getServices()` was inaccessible even when I changed the approach. Anyway it was throwing different weirdness for that 1 day only, maybe it was some cache, or some delay on their server, etc. 

TLDR I removed it now, should be good